builtin/logical/pki/path_acme_test.go

```diff
@@ -1405,6 +1428,131 @@ func TestAcmeValidationError(t *testing.T) {
+
+	leafKey, certs := doACMEWorkflow(t, vaultClient, acmeClient1)
+	acmeCert, err := x509.ParseCertificate(certs[0])
+	require.NoError(t, err, "failed parsing acme cert bytes")
+
+	// Make sure our cert is not revoked
+	certResp, err := vaultClient.Logical().ReadWithContext(ctx, "pki/cert/"+serialFromCert(acmeCert))
+	require.NoError(t, err, "failed to read certificate status")
+	require.NotNil(t, certResp, "certificate status response was nil")
+	revocationTime := certResp.Data["revocation_time"].(json.Number)
+	revocationTimeInt, err := revocationTime.Int64()
+	require.NoError(t, err, "failed converting revocation_time value: %v", revocationTime)
+	require.Equal(t, revocationTimeInt, int64(0),
+		"revocation time was not 0, cert was already revoked: %v", revocationTimeInt)
+
+	// Test that we can't revoke the certificate with another account's key
+	accountKey2, err := ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
+	require.NoError(t, err, "failed creating rsa key")
+
+	acmeClient2 := getAcmeClientForCluster(t, cluster, baseAcmeURL, accountKey2)
+	_, err = acmeClient2.Register(testCtx, &acme.Account{}, func(tosURL string) bool { return true })
+	require.NoError(t, err, "failed registering second account")
+
+	err = acmeClient2.RevokeCert(ctx, nil, certs[0], acme.CRLReasonUnspecified)
+	require.Error(t, err, "should have failed revoking the certificate with a different account")
+
+	// Make sure our cert is not revoked
+	certResp, err = vaultClient.Logical().ReadWithContext(ctx, "pki/cert/"+serialFromCert(acmeCert))
+	require.NoError(t, err, "failed to read certificate status")
+	require.NotNil(t, certResp, "certificate status response was nil")
+	revocationTime = certResp.Data["revocation_time"].(json.Number)
+	revocationTimeInt, err = revocationTime.Int64()
+	require.NoError(t, err, "failed converting revocation_time value: %v", revocationTime)
+	require.Equal(t, revocationTimeInt, int64(0),
+		"revocation time was not 0, cert was already revoked: %v", revocationTimeInt)
+
+	// But we can revoke if we sign the request with the certificate's key and a different account
+	err = acmeClient2.RevokeCert(ctx, leafKey, certs[0], acme.CRLReasonUnspecified)
+	require.NoError(t, err, "should have been allowed to revoke certificate with csr key across accounts")
+
+	// Make sure our cert is now revoked
+	certResp, err = vaultClient.Logical().ReadWithContext(ctx, "pki/cert/"+serialFromCert(acmeCert))
+	require.NoError(t, err, "failed to read certificate status")
+	require.NotNil(t, certResp, "certificate status response was nil")
+	revocationTime = certResp.Data["revocation_time"].(json.Number)
+	revocationTimeInt, err = revocationTime.Int64()
+	require.NoError(t, err, "failed converting revocation_time value: %v", revocationTime)
+	require.Greater(t, revocationTimeInt, int64(0),
+		"revocation time was not greater than 0, cert was not revoked: %v", revocationTimeInt)
+
+	// Make sure we can revoke a certificate without a registered ACME account
```

Comment:
isn't acmeClient1 a registered ACME account?  Or does that deregister when we revoke it's cert, above.